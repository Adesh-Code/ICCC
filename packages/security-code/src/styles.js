const styles = ({ tokens, primitives }) => ({
  "&.DhigSecurityCode": {
    width: "56px",
    height: "72px",
    borderTop: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`,
    borderBottom: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"][class*="MuiInputBase-inputAdornedEnd"]':
    {
      marginRight: "0"
    },
  "&.DhigSecurityCode[class*='MuiInput-root']:hover:before": {
    height: "2px",
    backgroundColor: tokens.colors["fill-engage"],
    boxShadow: "none",
    borderBottom: `${primitives.borderWidths.medium} solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode:hover": {
    boxShadow: "none"
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.primary.white}`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode input:focus": {
    boxShadow: `0px 2px 0px 0px ${tokens.colors["ink-focus"]} inset, 0px 0px 0px 2px ${tokens.colors["ink-focus"]} inset`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:not(.Mui-disabled)::before': {
    boxShadow: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]:before': {
    width: "56px"
  },
  "&.DhigSecurityCode input": {
    flexGrow: "1"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "56px",
    height: "72px",
    fontSize: `${primitives.fontSizeFallbacks["6"]}`,
    fontWeight: primitives.fontWeights.default
  },
  '&.DhigSecurityCode--dense [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "32px",
    height: "40px",
    fontSize: `${primitives.fontSizeFallbacks["3"]}`,
    fontWeight: primitives.fontWeights.default
  },
  "&.DhigSecurityCode--dense": {
    width: "32px",
    height: "40px"
  },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode--error, &.DhigSecurityCode:first-of-type.DhigSecurityCode--error, &.DhigSecurityCode:last-child.DhigSecurityCode--error, &.DhigSecurityCode--error:before":
    {
      borderColor: primitives.colors.accent.clay
    },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:after': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode:first-of-type": {
    borderLeft: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child": {
    borderRight: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child [class*='MuiDivider-root'], &.DhigSecurityCode:focus-within [class*='MuiDivider-root']":
    {
      display: "none"
    }
});

export default styles;
