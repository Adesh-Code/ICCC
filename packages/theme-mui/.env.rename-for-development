# To load the MUI X Pro license for local development, rename this file to .env and replace with the actual license string
DHIG_MUIXPRO_ENTRY=PASTE_MUI_X_PRO_LICENSE_STRING_HERE