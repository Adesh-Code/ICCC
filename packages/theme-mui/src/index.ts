// Themes
export { default as themeDark } from "./themes/dark";
export { default as themeLight } from "./themes/light";

// Utilities
export { configureClassnameGenerator } from "./classname-generator";
export { setLicenseKey } from "./set-license-key";
