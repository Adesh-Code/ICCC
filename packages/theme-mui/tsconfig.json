{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "lib/esm",
    "resolveJsonModule": true,
    "declarationDir": "lib/types"
  },
  "include": ["src"]
}
