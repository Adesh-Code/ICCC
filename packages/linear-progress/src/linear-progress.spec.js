import React from "react";
import LinearProgress, { LinearProgressLabel } from "./linear-progress";
import { createSerializer, matchers } from "@emotion/jest";
import { screen, render } from "@testing-library/react";
import { themeLight } from "@iccc/theme-mui";
import { createTheme, ThemeProvider } from "@mui/material/styles";

const light = createTheme(themeLight);

expect.extend(matchers);
expect.addSnapshotSerializer(createSerializer({ includeStyles: false }));

const arrange = (args) => {
  return render(
    <ThemeProvider theme={light}>
      <LinearProgress {...args} />
    </ThemeProvider>
  );
};

describe("Linear progress", () => {
  it("renders a linear progress", () => {
    const { container } = arrange();
    expect(container).toMatchSnapshot();
  });

  it("renders the linear progress in medium size", () => {
    arrange({
      size: "medium"
    });
    const progress = screen.getByRole("progressbar");
    expect(progress).toHaveClass("DhigLinearProgress--size--medium");
  });

  it("renders percentage for determinate variant", () => {
    const { container } = arrange({
      variant: "determinate",
      value: 25,
      children: (
        <LinearProgressLabel
          variant="determinate"
          label="Determinate label"
          value={25}
        />
      )
    });
    expect(container).toHaveTextContent("25%");
    expect(container).toHaveTextContent("Determinate label");
  });

  it("renders label with percentage", () => {
    const { container } = arrange({
      variant: "determinate",
      value: 25,
      children: <LinearProgressLabel variant="determinate" value={25} />
    });
    expect(container).toHaveTextContent("25%");
  });

  it("accepts classnames", () => {
    arrange({
      className: "foo"
    });
    const progress = screen.getAllByRole("progressbar")[0];
    expect(progress).toHaveClass("foo");
  });
});
