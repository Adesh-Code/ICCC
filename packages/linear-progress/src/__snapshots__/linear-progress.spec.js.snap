// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Linear progress renders a linear progress 1`] = `
<div>
  <span
    class="MuiLinearProgress-root icccLinearProgress--size--small MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate emotion-0"
    role="progressbar"
  >
    <span
      class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate emotion-1"
    />
    <span
      class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar2Indeterminate emotion-2"
    />
  </span>
</div>
`;
